File: go-sdk/pkg/models/shared/article.go
package shared

type ArticleArticle struct {
	BodyMarkdown   *string `json:"body_markdown,omitempty"`
	CanonicalURL   *string `json:"canonical_url,omitempty"`
	Description    *string `json:"description,omitempty"`
	MainImage      *string `json:"main_image,omitempty"`
	OrganizationID *int64  `json:"organization_id,omitempty"`
	Published      *bool   `json:"published,omitempty"`
	Series         *string `json:"series,omitempty"`
	Tags           *string `json:"tags,omitempty"`
	Title          *string `json:"title,omitempty"`
}

// Article
// Representation of an Article to be created/updated
type Article struct {
	Article *ArticleArticle `json:"article,omitempty"`
}

File: ts-sdk/src/sdk/models/shared/article.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class ArticleArticle extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=body_markdown" })
  bodyMarkdown?: string;

  @SpeakeasyMetadata({ data: "json, name=canonical_url" })
  canonicalUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=main_image" })
  mainImage?: string;

  @SpeakeasyMetadata({ data: "json, name=organization_id" })
  organizationId?: number;

  @SpeakeasyMetadata({ data: "json, name=published" })
  published?: boolean;

  @SpeakeasyMetadata({ data: "json, name=series" })
  series?: string;

  @SpeakeasyMetadata({ data: "json, name=tags" })
  tags?: string;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;
}


// Article
/** 
 * Representation of an Article to be created/updated
**/
export class Article extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=article" })
  article?: ArticleArticle;
}
